spirv_msl.cpp

```diff
@@ -3646,6 +3707,704 @@ void CompilerMSL::emit_custom_functions()
 			statement("");
 			break;
 
+		case SPVFuncImplChromaReconstructNearest2Plane:
+			statement("template<typename T, typename... LodOptions>");
+			statement("inline vec<T, 4> spvChromaReconstructNearest(texture2d<T> plane0, texture2d<T> plane1, sampler "
```

Comment:
I don't follow most of these implementations. Care to elaborate how these implementations are found?

Reply:
They're based on [the descriptions of explicit chroma reconstruction in the Vulkan standard](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/chap15.html#textures-explicit-reconstruction).

Reply:
I see. Sounds good then.